includes/admin/class-wc-rest-payments-payment-intents-controller.php

```diff
@@ -46,4 +74,47 @@ public function get_payment_intent( $request ) {
 
 		return $this->forward_request( 'get_intent', [ $payment_intent_id ] );
 	}
+
+	/**
+	 * Create a payment intent.
+	 *
+	 * @param WP_REST_Request $request data about the request.
+	 */
+	public function post_payment_intent( $request ) {
+		try {
+
+			$order_id = $request->get_param( 'order_id' );
+			$order    = wc_get_order( $order_id );
+			$currency = strtolower( $order->get_currency() );
+			$amount   = WC_Payments_Utils::prepare_amount( $order->get_total(), $currency );
+			$name     = sanitize_text_field( $order->get_billing_first_name() ) . ' ' . sanitize_text_field( $order->get_billing_last_name() );
+			$email    = sanitize_email( $order->get_billing_email() );
+			$metadata = [
+				'customer_name'  => $name,
+				'customer_email' => $email,
+				'site_url'       => esc_url( get_site_url() ),
+				'order_id'       => $order->get_id(),
+				'order_number'   => $order->get_order_number(),
+				'order_key'      => $order->get_order_key(),
+				'payment_type'   => 'single',
+			];
+
```

Comment:
It might be that you missed this comment in my previous review 
p1690350125782319/1690349903.234759-slack-C01BPL3ALGP

> `metadata` are built for `$wcpay_server_request` right here. However, we have already have a method for that purpose `WC_Payment_Gateway_WCPay::get_metadata_from_order` - which is currently `protected` however we can change it to `public`. And ideally, all of these would become Util methods, but this is not a focus of this PR.

Reply:
Done